### PR TITLE
fixe image size

### PR DESCRIPTION
**Pull request by fahedouch**
fixing https://github.com/containerd/nerdctl/issues/777
`ImageSize` is the size of the unpacked snapshots. for multi-platform image, snapshot host single platform layer(s). So the computing of image size shouldn't be based on image platform

Signed-off-by: fahed dorgaa <fahed.dorgaa@gmail.com>

### CODE REVIEW

**Review by AkihiroSuda — CHANGES_REQUESTED** [review]
This only shows the size of the default platform

File: cmd/nerdctl/images.go [hunk 1]
```diff
@@ -314,8 +314,8 @@ func (key snapshotKey) add(ctx context.Context, s snapshots.Snapshotter, usage *
 
 // unpackedImageSize is the size of the unpacked snapshots.
 // Does not contain the size of the blobs in the content store. (Corresponds to Docker).
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:316]: Maybe we should show both the blob size and the unpacked size to avoid confusion for multi-platform images

**Comment by fahedouch** [comment]
related issue https://github.com/containerd/nerdctl/issues/796

File: cmd/nerdctl/images.go (hunk 1 above)
Comment (fahedouch) [reply cmd/nerdctl/images.go]: fixed

File: cmd/nerdctl/images.go [hunk 2]
```diff
@@ -109,7 +109,8 @@ type imagePrintable struct {
 	ID           string // image target digest (not config digest, unlike Docker), or its short form
 	Repository   string
 	Tag          string // "<none>" or tag
-	Size         string
+	UnpackedSize string // the size of the unpacked snapshots.
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:112]: the `Size` field needs to be retained for Docker compatibility

File: cmd/nerdctl/images.go [hunk 3]
```diff
@@ -142,9 +143,9 @@ func printImages(ctx context.Context, cmd *cobra.Command, client *containerd.Cli
 		w = tabwriter.NewWriter(w, 4, 8, 4, ' ', 0)
 		if !quiet {
 			if digestsFlag {
-				fmt.Fprintln(w, "REPOSITORY\tTAG\tDIGEST\tIMAGE ID\tCREATED\tPLATFORM\tSIZE")
+				fmt.Fprintln(w, "REPOSITORY\tTAG\tDIGEST\tIMAGE ID\tCREATED\tPLATFORM\tSIZE\tBLOB_SIZE")
 			} else {
-				fmt.Fprintln(w, "REPOSITORY\tTAG\tIMAGE ID\tCREATED\tPLATFORM\tSIZE")
+				fmt.Fprintln(w, "REPOSITORY\tTAG\tIMAGE ID\tCREATED\tPLATFORM\tSIZE\tBLOB_SIZE")
```
Comment (AkihiroSuda) [inline cmd/nerdctl/images.go:148]: Do we use underscore in other places?

**Review by AkihiroSuda — APPROVED** [review]
Thanks

**Comment by AkihiroSuda** [comment]
`nerdctl images` might be slower than before (which has been already slow).

We should do some benchmark and probably have `--quick` (`--fast`?, idk) flag to skip slow size calculation

**Comment by fahedouch** [comment]
@AkihiroSuda I think we can merge this, I will open a new PR!

**Pull request merged**